File: framework-log/src/main/java/com/example/framework/log/entity/dto/LogRecordDto.java
package com.example.framework.log.entity.dto;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.validator.constraints.Length;

import java.io.Serializable;
import java.util.Date;

/**
 * @author MaGuangZu
 * @since 2022-08-19
 */
@EqualsAndHashCode(callSuper = false)
@Setter
@Getter
public class LogRecordDto implements Serializable {

	private String id;
	private String tenant;

	@Length(max = 200)
	private String type;
	private String subType;

	@Length(max = 200)
	private String bizNo;

	@Length(max = 63)
	private String operator;

	@Length(max = 511)
	private String action;
	private boolean fail = false;
	private Date createTime;
	private String extra;
}

File: framework-log/src/main/java/com/example/framework/log/entity/dto/LogRecordDto.kt
package com.example.framework.log.entity.dto

import org.hibernate.validator.constraints.Length
import org.springframework.data.domain.PageRequest
import org.springframework.data.domain.Sort
import java.io.Serializable
import java.util.*

/**
 * @author MaGuangZu
 * @since 2022-08-19
 */
class LogRecordDto(page: Int, size: Int, sort: Sort?) : PageRequest(page, size, sort), Serializable {
    var id: String? = null
    var tenant: String? = null
    var type: @Length(max = 200, message = "type max length is 200") String? = null
    var subType: String? = null
    var bizNo: @Length(max = 200, message = "bizNo max length is 200") String? = null
    var operator: @Length(max = 63, message = "operator max length 63") String? = null
    var action: @Length(max = 511, message = "operator max length 511") String? = null
    var fail = false
    var createTime: Date? = null
    var extra: String? = null
}
